src/module/actor/creature/index.ts

```diff
@@ -303,6 +303,9 @@ export abstract class CreaturePF2e extends ActorPF2e {
             };
         }
 
+        // Always add the actor's ID to the rolloptions.
+        this.rollOptions.all[`self:id:${this.id}`] = true;
```

Comment:
This _could_ be moved to `ActorPF2e`, as it doesn't cause any problems, but that doesn't bring any new features with it.
Hazards and loot boxes won't have their ID grabbed for `target:id`, and the hazards' rolls are uninspectable due to an error anyways. Loot boxes don't usually roll.